test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+import inspect
+import torch
+import unittest
+from unittest.mock import MagicMock, patch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+
+
+# Returns a database of args & kwargs that can be used to construct each module.
+# Each entry is in class -> (args, kwargs) format.
+# Example: torch.nn.Linear -> ([10, 5], {})
+def build_constructor_arg_db():
+    return {
+        torch.nn.AdaptiveAvgPool1d: ([5], {}),
+        torch.nn.AdaptiveAvgPool2d: ([5], {}),
+        torch.nn.AdaptiveAvgPool3d: ([5], {}),
+        torch.nn.AdaptiveLogSoftmaxWithLoss: ([100, 20, [5, 10, 15]], {}),
+        torch.nn.AdaptiveMaxPool1d: ([5], {}),
```

Comment:
Style nit (would not block PR on this or make this change if it's a nontrivial amount of work): prefer tuples to lists

Reply:
Any particular reason? I don't mind doing the work to make the change, but it makes it harder to read imo